### PR TITLE
[wip] Enable core dumping and emit stack traces in the CI

### PR DESCRIPTION
**Pull request by AndrejMitrovic**

### CODE REVIEW

File: ci/ci_osx_setup.sh [hunk 1]
```diff
@@ -7,3 +7,8 @@ mkdir -p $HOME/Dependencies/
 wget -P $HOME/Dependencies/ https://homebrew.bintray.com/bottles/libsodium-1.0.18.high_sierra.bottle.tar.gz
 tar -C /usr/local/Cellar/ -xf $HOME/Dependencies/libsodium-1.0.18.high_sierra.bottle.tar.gz
 brew link libsodium
+
+# enable core dumps
+ulimit -c unlimited -S
+sudo mkdir /cores/ && sudo chmod 777 /cores/
+sudo sysctl -w kern.corefile=/cores/%E.%p
```
Comment (Geod24) [inline ci/ci_osx_setup.sh:14]: Works on OSX as well ?

File: ci/ci_linux_setup.sh [hunk 2]
```diff
@@ -16,3 +16,8 @@ make -j4
 sudo make install
 sudo ldconfig # Refresh cache
 popd
+
+# enable core dumps
+ulimit -c unlimited -S
+sudo mkdir /cores/ && sudo chmod 777 /cores/
+sudo sysctl -w kernel.core_pattern=/tmp/core-%e.%p
```
Comment (Geod24) [inline ci/ci_linux_setup.sh]: I thought we'd want `%E` here ?

File: ci/core-dump.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -veo pipefail
+
+for file in /cores/*; do
+    echo "Core file: $file"
+    DUB_EXEC=$(echo "$file" | sed 's/\/cores\///' | sed -E 's/\.[0-9]+$//' | tr '!' '/')
```
Comment (Geod24) [inline ci/core-dump.sh]: Maybe renamed this variable ?

File: ci/core-dump.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    echo "Executable: $DUB_EXEC"
```
Comment (Geod24) [inline ci/core-dump.sh]: I'd merge both `echo` not to clog the output

File: ci/core-dump.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    sudo lldb -c $file -o "bt all" -o "script import os; os._exit(1)"
```
Comment (Geod24) [inline ci/core-dump.sh:7]: Do we get a recent `lldb` ? They added `--batch` support too.

https://stackoverflow.com/a/36580453/1021344

**Comment by AndrejMitrovic** [comment]
I'll get back to this soon, CI testing is fun you know.

**Comment by Geod24** [comment]
How soon @AndrejMitrovic ? :P

**Comment by AndrejMitrovic** [comment]
I found something better, but it seems it's only supported on Linux. On the vibe.d docs at the very end: https://vibed.org/docs#advanced

```
import etc.linux.memoryerror;
static if (is(typeof(registerMemoryErrorHandler)))
	registerMemoryErrorHandler();
```

I'm trying it out now.

**Comment by AndrejMitrovic** [comment]
I've made a backup of the old branch in case it's worth working on it: https://github.com/AndrejMitrovic/agora/tree/backup-stack-traces

**Comment by AndrejMitrovic** [comment]
Ah that doesn't seem to change anything. The stack traces are there on Linux, it's really just OSX that's the issue I think..

**Comment by AndrejMitrovic** [comment]
Ok resumed testing ideas.

**Comment by AndrejMitrovic** [comment]
This is stale, and I never managed to get it working right. Will let others work on it if they have the time.

**Pull request closed without merging**